internal/util/filesystem.go

```diff
@@ -61,8 +66,51 @@ func IsHidden(fileName string) bool {
 	}
 }
 
-func IsZip(fileName string) bool {
-	return strings.HasSuffix(fileName, ".zip")
+func getFsFromTar(r io.Reader) fs.FS {
+	tfs, err := tarfs.New(r)
+	if err != nil {
+		panic(fmt.Errorf("failed to open tar file: %w", err))
```

Comment:
I know it makes the `OpenDirectoryFS` function a bit awkward to use. But, because we are dealing with files and there's good probability that things could go wrong, we should return the errors back to the caller and exit gracefully rather than panicking.